allennlp/sanity_checks/task_checklists/textual_entailment_suite.py

```diff
@@ -0,0 +1,348 @@
+from typing import Optional, Tuple, Iterable
+from allennlp.sanity_checks.task_checklists.task_suite import TaskSuite
+from checklist.test_suite import TestSuite
+from checklist.test_types import MFT
+from checklist.perturb import Perturb
+import itertools
+import numpy as np
+from allennlp.sanity_checks.task_checklists import utils
+from overrides import overrides
+
+
+def wrap_apply_to_each(fn, both=False, *args, **kwargs):
+    """
+    Wraps the perturb function so that it is applied to
+    both elements in the (premise, hypothesis) tuple.
+    """
+
+    def new_fn(pair, *args, **kwargs):
+        premise, hypothesis = pair
+        ret = []
+        fn_premise = fn(premise, *args, **kwargs)
+        fn_hypothesis = fn(hypothesis, *args, **kwargs)
+        if type(fn_premise) != list:
+            fn_premise = [fn_premise]
+        if type(fn_hypothesis) != list:
+            fn_hypothesis = [fn_hypothesis]
+        ret.extend([(x, str(hypothesis)) for x in fn_premise])
+        ret.extend([(str(premise), x) for x in fn_hypothesis])
+        if both:
+            ret.extend([(x, x2) for x, x2 in itertools.product(fn_premise, fn_hypothesis)])
+        return [x for x in ret if x[0] and x[1]]
```

Comment:
Is this trying to guard against the empty string? Why would the perturbations ever be empty?

Reply:
They can be, if no perturbation can be applied. Eg. if the sentence is "This is a good movie", a perturbation which toggles contractions will have no effect. I've added a comment.